### PR TITLE
 Replace usages of `ASM6` with `ASM_LEVEL`

### PR DESCRIPTION
**Pull request by bamboo**

### CODE REVIEW

File: subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt [hunk 1]
```diff
@@ -1141,6 +1145,70 @@ class ProjectSchemaAccessorsIntegrationTest : AbstractPluginIntegrationTest() {
...
+        assertThat(
+            build("-q").output,
+            containsString("build.Java11Plugin${'$'}Java11Extension")
+        )
+    }
+
+    private
+    fun assumeJava11() {
+        assumeTrue("Test requires Java 11", JavaVersion.current().isJava11Compatible)
+    }
```
Comment (eskatos) [inline subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt]: :o: I'd move this close to the existing assumptions here 

https://github.com/gradle/gradle/blob/611346be9533b5ffc4de03234991d9958449aa12/subprojects/kotlin-dsl-test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/AbstractKotlinIntegrationTest.kt#L190-L198

**Comment by bamboo** [comment]
FTR, I did include `GradleApiParameterNamesTransform` in my first iteration but it caused the build to fail so I removed it.

File: subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt (hunk 1 above)
Comment (bamboo) [reply subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/ProjectSchemaAccessorsIntegrationTest.kt]: Done in 24dce2a.

**Comment by bamboo** [comment]
[Ready for Release](https://builds.gradle.org/viewLog.html?buildId=20359646&) ✅

**Pull request merged**